pkg/operator/observe_config.go

```diff
@@ -73,9 +78,29 @@ func (c ConfigObserver) sync() error {
 		unstructured.SetNestedField(observedConfig, val, "projectConfig", "defaultNodeSelector")
 	}
 
+	// we read the etcd endpoints from the endpoints object and then manually pull out the hostnames to get the etcd urls for our config.
+	// Setting them observed config causes the normal reconciliation loop to run
+	etcdURLs := []string{}
+	etcdEndpoints, err := c.kubeClient.CoreV1().Endpoints(etcdNamespaceName).Get("etcd", metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+	if etcdEndpoints != nil {
+		for _, subset := range etcdEndpoints.Subsets {
+			for _, address := range subset.Addresses {
+				etcdURLs = append(etcdURLs, "https://"+address.Hostname+"."+etcdEndpoints.Annotations["alpha.installer.openshift.io/dns-suffix"]+":2379")
```

Comment:
without the annotation we get a trailing dot. Intended?

Reply:
> without the annotation we get a trailing dot. Intended?

Without the annotation I think we fail in general.